An error occurred while calling method 'EventSubscriber.receive()'

OH snapshot 1361 (same on 1360 too)
ESH 0.10.0.201809111909 

When the Auto Update option has not been selected for a managed item, the following error will occur when manipulating the item, and any linked channels are not updated. An item can be created without choosing an Auto Update option, and it's a new field, so previously created items would not have it populated either.
```
2018-09-13 12:48:56.985 [ERROR] [org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler] - An error occurred while calling method 'EventSubscriber.receive()' on 'org.eclipse.smarthome.core.thing.internal.CommunicationManager@ad062db': null
java.lang.NullPointerException: null
        at org.eclipse.smarthome.core.thing.internal.AutoUpdateManager.receiveCommand(AutoUpdateManager.java:141) [114:org.eclipse.smarthome.core.thing:0.10.0.201809111909]
        at org.eclipse.smarthome.core.thing.internal.CommunicationManager.receiveCommand(CommunicationManager.java:266) [114:org.eclipse.smarthome.core.thing:0.10.0.201809111909]
        at org.eclipse.smarthome.core.thing.internal.CommunicationManager.receive(CommunicationManager.java:145) [114:org.eclipse.smarthome.core.thing:0.10.0.201809111909]
        at sun.reflect.GeneratedMethodAccessor16.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153) [108:org.eclipse.smarthome.core:0.10.0.201809111909]
        at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [108:org.eclipse.smarthome.core:0.10.0.201809111909]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
2018-09-13 12:48:57.003 [ERROR] [org.eclipse.smarthome.core.internal.events.EventHandler] - Dispatching/filtering event for subscriber 'org.eclipse.smarthome.core.events.EventSubscriber' failed: null
java.lang.NullPointerException: null
        at org.eclipse.smarthome.core.thing.internal.AutoUpdateManager.receiveCommand(AutoUpdateManager.java:141) [114:org.eclipse.smarthome.core.thing:0.10.0.201809111909]
        at org.eclipse.smarthome.core.thing.internal.CommunicationManager.receiveCommand(CommunicationManager.java:266) [114:org.eclipse.smarthome.core.thing:0.10.0.201809111909]
        at org.eclipse.smarthome.core.thing.internal.CommunicationManager.receive(CommunicationManager.java:145) [114:org.eclipse.smarthome.core.thing:0.10.0.201809111909]
        at sun.reflect.GeneratedMethodAccessor16.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153) [108:org.eclipse.smarthome.core:0.10.0.201809111909]
        at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [108:org.eclipse.smarthome.core:0.10.0.201809111909]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```
More discussion here...
https://community.openhab.org/t/new-zwave-device-fails-attempts-to-fix-leads-to-all-new-thing-names-breaking-item-links-and-rules/51326/25?u=5iver

Possibly related to this issue, items were very laggy after an OH restart, and then all became unresponsive. After about 15 minutes, everything came back and responded as normal. Another user reported the same in the forum thread linked above.

--- a/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/dto/MetadataDTO.java
+++ b/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/dto/MetadataDTO.java
@@ -14,18 +14,19 @@ package org.eclipse.smarthome.core.items.dto;
 
 import java.util.Map;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  * This is a data transfer object that is used to serialize metadata for a certain namespace and item.
  *
  * @author Kai Kreuzer - Initial contribution and API
  *
  */
+@NonNullByDefault
 public class MetadataDTO {
 
-    public String value;
-    public Map<String, Object> config;
-
-    public MetadataDTO() {
-    }
+    public @Nullable String value;
+    public @Nullable Map<String, Object> config;
 
 }

--- a/bundles/io/org.eclipse.smarthome.io.rest.core.test/src/test/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResourceOSGiTest.java
+++ b/bundles/io/org.eclipse.smarthome.io.rest.core.test/src/test/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResourceOSGiTest.java
@@ -232,6 +232,22 @@ public class ItemResourceOSGiTest extends JavaOSGiTest {
     }
 
     @Test
+    public void testAddMetadata_ValueEmtpy() {
+        MetadataDTO dto = new MetadataDTO();
+        dto.value = "";
+        Response response = itemResource.addMetadata(ITEM_NAME1, "foo", dto);
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test
+    public void testAddMetadata_ValueNull() {
+        MetadataDTO dto = new MetadataDTO();
+        dto.value = null;
+        Response response = itemResource.addMetadata(ITEM_NAME1, "foo", dto);
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test
     public void testAddMetadata_update() {
         MetadataDTO dto = new MetadataDTO();
         dto.value = "some value";

--- a/bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java
+++ b/bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java
@@ -544,6 +544,7 @@ public class ItemResource implements RESTResource {
     @ApiResponses(value = { //
             @ApiResponse(code = 200, message = "OK"), //
             @ApiResponse(code = 201, message = "Created"), //
+            @ApiResponse(code = 400, message = "Metadata value empty."), //
             @ApiResponse(code = 404, message = "Item not found."), //
             @ApiResponse(code = 405, message = "Metadata not editable.") })
     public Response addMetadata(@PathParam("itemname") @ApiParam(value = "item name", required = true) String itemname,
@@ -557,8 +558,15 @@ public class ItemResource implements RESTResource {
             return Response.status(Status.NOT_FOUND).build();
         }
 
+        String value = metadata.value;
+        if (value == null || value.isEmpty()) {
+            logger.info("Received HTTP PUT request at '{}' for item '{}' with empty metadata.", uriInfo.getPath(),
+                    itemname);
+            return Response.status(Status.BAD_REQUEST).build();
+        }
+
         MetadataKey key = new MetadataKey(namespace, itemname);
-        Metadata md = new Metadata(key, metadata.value, metadata.config);
+        Metadata md = new Metadata(key, value, metadata.config);
         if (metadataRegistry.get(key) == null) {
             metadataRegistry.add(md);
             return Response.status(Status.CREATED).type(MediaType.TEXT_PLAIN).build();

--- a/bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java
+++ b/bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/item/ItemResource.java
@@ -48,7 +48,6 @@ import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.smarthome.core.auth.Role;
 import org.eclipse.smarthome.core.events.EventPublisher;
-import org.eclipse.smarthome.core.items.ActiveItem;
 import org.eclipse.smarthome.core.items.GenericItem;
 import org.eclipse.smarthome.core.items.GroupItem;
 import org.eclipse.smarthome.core.items.Item;
@@ -504,7 +503,7 @@ public class ItemResource implements RESTResource {
             return Response.status(Status.METHOD_NOT_ALLOWED).build();
         }
 
-        ((ActiveItem) item).addTag(tag);
+        ((GenericItem) item).addTag(tag);
         managedItemProvider.update(item);
 
         return Response.ok(null, MediaType.TEXT_PLAIN).build();
@@ -530,7 +529,7 @@ public class ItemResource implements RESTResource {
             return Response.status(Status.METHOD_NOT_ALLOWED).build();
         }
 
-        ((ActiveItem) item).removeTag(tag);
+        ((GenericItem) item).removeTag(tag);
         managedItemProvider.update(item);
 
         return Response.ok(null, MediaType.TEXT_PLAIN).build();
@@ -630,7 +629,7 @@ public class ItemResource implements RESTResource {
     public Response createOrUpdateItem(
             @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value = "language") String language,
             @PathParam("itemname") @ApiParam(value = "item name", required = true) String itemname,
-            @ApiParam(value = "item data", required = true) GroupItemDTO item) {
+            @ApiParam(value = "item data", required = true) @Nullable GroupItemDTO item) {
         final Locale locale = localeService.getLocale(language);
 
         // If we didn't get an item bean, then return!
@@ -674,7 +673,8 @@ public class ItemResource implements RESTResource {
     @ApiOperation(value = "Adds a list of items to the registry or updates the existing items.")
     @ApiResponses(value = { @ApiResponse(code = 200, message = "OK", response = String.class),
             @ApiResponse(code = 400, message = "Item list is null.") })
-    public Response createOrUpdateItems(@ApiParam(value = "array of item data", required = true) GroupItemDTO[] items) {
+    public Response createOrUpdateItems(
+            @ApiParam(value = "array of item data", required = true) GroupItemDTO @Nullable [] items) {
         // If we didn't get an item list bean, then return!
         if (items == null) {
             return Response.status(Status.BAD_REQUEST).build();

--- a/extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/items/controllers.items.js
+++ b/extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/items/controllers.items.js
@@ -235,7 +235,7 @@ angular.module('PaperUI.items')//
             var resolveLater = false;
             if (!originalItem.metadata || JSON.stringify(metadata) !== JSON.stringify(originalItem.metadata)) {
                 for ( var namespace in metadata) {
-                    if (!metadata.hasOwnProperty(namespace)) {
+                    if (!metadata.hasOwnProperty(namespace) || metadata[namespace].value === undefined || metadata[namespace].value.length === 0) {
                         continue;
                     }
 

--- a/extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/items/controllers.items.js
+++ b/extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/items/controllers.items.js
@@ -233,6 +233,7 @@ angular.module('PaperUI.items')//
     function updateMetadata(itemName, metadata, originalItem, toastText) {
         return $q(function(resolve, reject) {
             var resolveLater = false;
+            var namespaceUpdatePromisses = [];
             if (!originalItem.metadata || JSON.stringify(metadata) !== JSON.stringify(originalItem.metadata)) {
                 for ( var namespace in metadata) {
                     if (!metadata.hasOwnProperty(namespace) || metadata[namespace].value === undefined || metadata[namespace].value.length === 0) {
@@ -241,21 +242,23 @@ angular.module('PaperUI.items')//
 
                     if (!originalItem.metadata || JSON.stringify(metadata[namespace]) !== JSON.stringify(originalItem.metadata[namespace])) {
                         resolveLater = true;
-                        itemService.updateMetadata({
+                        var namespaceUpdatePromise = itemService.updateMetadata({
                             itemName : itemName,
                             namespace : namespace
-                        }, metadata[namespace], function() {
+                        }, metadata[namespace]).$promise.then(function() {
                             if (toastText) {
                                 toastService.showDefaultToast(toastText);
                             }
-
-                            resolve();
                         });
+
+                        namespaceUpdatePromisses.push(namespaceUpdatePromise);
                     }
                 }
             }
             if (!resolveLater) {
                 resolve();
+            } else {
+                $q.all(namespaceUpdatePromisses).then(resolve());
             }
         });
     }

